nixos/modules/services/networking/aria2.nix

```diff
@@ -117,6 +120,7 @@ in
           touch "${sessionFile}"
         fi
         cp -f "${settingsFile}" "${settingsDir}/aria2.conf"
+        echo "rpc-secret=$(cat ${escapeShellArg cfg.rpcSecretFile})" >> "${settingsDir}/aria2.conf"
```

Comment:
Note: It might be worth to replace `${settingsDir}` with `$RUNTIME_DIRECTORY` and to set

```
RuntimeDirectory = "aria2";
RuntimeDirectoryMode = "0700";
```

systemd will clean up the file when the service shut down.

Reply:
Given that the module has no tests, I would rather avoid making significant changes like that.

And I don't really know enough about aria to add a test, this was mostly a drive-by PR as I was evaluating various download clients.

Reply:
It should be done anyway because this PR breaks aria2 for me:

```bash
Feb 09 12:38:13 grandlibrary systemd[1]: Starting aria2 Service...
Feb 09 12:38:13 grandlibrary aria2-pre-start[16680]: /nix/store/1wi2jch5hqsw6cp2k0nl6v46f5n3hdpx-unit-script-aria2-pre-start/bin/aria2-pre-start: line 8: /var/lib/aria2/aria2.conf: Permission denied
Feb 09 12:38:13 grandlibrary systemd[1]: aria2.service: Control process exited, code=exited, status=1/FAILURE
Feb 09 12:38:13 grandlibrary systemd[1]: aria2.service: Failed with result 'exit-code'.
Feb 09 12:38:13 grandlibrary systemd[1]: Failed to start aria2 Service.
```

Reply:
Try `chown`ing the file to aria? And check that it has write permissions.

Reply:
Ok yeah, that was it; for some reason it was read-only.